cmake/ClangFormat.cmake

```diff
@@ -0,0 +1,90 @@
+function(clang_format_setup)
+  cmake_parse_arguments(
+        CF_ARG # prefix 
```

Comment:
It seems a little inconsistent to me that the prefix is `CF_ARG` but the exported executable is `CLANG_FORMAT_EXECUTABLE`. I'm having a hard time finding useful examples on the internet, does this reasonably follow CMake conventions @jslee02?

Reply:
CMake is not a good tool to find a good convention of using it. See this [list](https://github.com/onqtam/awesome-cmake). It's horrific. :scream:

Back to the question, it seems people uses lower caess for the prefix (e.g., here and here). Also, there is tendency of using upper cases for global variables and lower cases for local variables. By the way, `CLANG_FORMAT_EXECUTABLE` is set by `find_program` not by `cmake_parse_arguments`.